network-area-diagram/src/main/java/com/powsybl/nad/NetworkAreaDiagram.java

```diff
@@ -35,116 +31,91 @@
     }
 
-    public void draw(Path svgFile, SvgParameters svgParameters, LayoutParameters layoutParameters) {
-        draw(svgFile, svgParameters, layoutParameters, new TopologicalStyleProvider(network));
+    public static void draw(Network network, Path svgFile, List<String> voltageLevelIds) {
+        draw(network, svgFile, new ParamBuilder().build(), VoltageLevelFilter.createVoltageLevelsFilter(network, voltageLevelIds));
     }
 
-    public void draw(Path svgFile, SvgParameters svgParameters, LayoutParameters layoutParameters,
-                                   StyleProvider styleProvider) {
-        draw(svgFile, svgParameters, layoutParameters, styleProvider, new DefaultLabelProvider(network, svgParameters));
+    public static void draw(Network network, Path svgFile, List<String> voltageLevelIds, int depth) {
+        draw(network, svgFile, new ParamBuilder().build(), VoltageLevelFilter.createVoltageLevelsDepthFilter(network, voltageLevelIds, depth));
     }
 
-    public void draw(Path svgFile, SvgParameters svgParameters, LayoutParameters layoutParameters,
-                                   StyleProvider styleProvider, LabelProvider labelProvider) {
-        draw(svgFile, svgParameters, layoutParameters, styleProvider, labelProvider, new BasicForceLayoutFactory());
-    }
+    /* ---------------------------------------------------------------- */
+    // Network, Path, Param, Predicate
+    /* ---------------------------------------------------------------- */
 
-    public void draw(Path svgFile, SvgParameters svgParameters, LayoutParameters layoutParameters,
-                                   StyleProvider styleProvider, LabelProvider labelProvider, LayoutFactory layoutFactory) {
-        draw(svgFile, svgParameters, layoutParameters, styleProvider, labelProvider, layoutFactory, new IntIdProvider());
-    }
+    public static void draw(Network network, Path svgFile, Param param, Predicate<VoltageLevel> voltageLevelFilter) {
 
-    public void draw(Path svgFile, SvgParameters svgParameters, LayoutParameters layoutParameters,
-                                   StyleProvider styleProvider, LabelProvider labelProvider, LayoutFactory layoutFactory,
-                                   IdProvider idProvider) {
+        Objects.requireNonNull(network);
         Objects.requireNonNull(svgFile);
+        LayoutParameters layoutParameters = param.getLayoutParameters();
+        SvgParameters svgParameters = param.getSvgParameters();
+        StyleProvider styleProvider = param.createStyleProvider(network);
+        LayoutFactory layoutFactory = param.getLayoutFactory();
+        IdProvider idProvider = param.createIdProvider();
+        LabelProvider labelProvider = param.createLabelProvider(network);
         Objects.requireNonNull(layoutParameters);
         Objects.requireNonNull(svgParameters);
         Objects.requireNonNull(styleProvider);
         Objects.requireNonNull(layoutFactory);
         Objects.requireNonNull(idProvider);
```

Comment:
In the Param constructor, those checks are on :
- layoutParameters (same as before)
- svgParameters (same as before)
- layoutFactory (same as before)
- styleProviderFactory (instead of styleProvider)
- idProviderFactory (instead of idProvider)